### PR TITLE
support  mysqlNodePort and statusNodePort

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
resolve https://github.com/pingcap/tidb-operator/issues/2915. support user specied `mysqlNodePort` and `statusNodePort`

### What is changed and how does it work?
1. Add `MySQLNodePort` and `StatusNodePort` field to support specified `mysqlNodePort` and `statusNodePort.
2. Don't retain old svc config if `MySQLNodePort` or `StatusNodePort` is set.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
    Test manually set `mySQLNodePort` and `statusNodePort`. They can work in svc. Besides, if we don't set them, the svc port won't change every time.

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support specifying mysqlNodePort and statusNodePort for TiDB Service Spec.
```

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -240,6 +240,13 @@ func RetainManagedFields(desiredSvc, existedSvc *corev1.Service) {
 	// Retain NodePorts
 	for id, dport := range desiredSvc.Spec.Ports {
 		for _, eport := range existedSvc.Spec.Ports {
+			if tidbSvcSpec != nil {
+				if dport.Name == "mysql-client" && tidbSvcSpec.GetMySQLNodePort() != 0 {
```
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply pkg/util/util.go:244]: BTW, this should be a common function that may be called by other service sync functions and may also add other fields (even if not now), is it better to handle this out of this function? WDYT @lichunzhu @cofyc

File: examples/basic/tidb-cluster.yaml [hunk 2]
```diff
@@ -33,5 +33,8 @@ spec:
     baseImage: pingcap/tidb
     replicas: 1
     service:
-      type: ClusterIP
+      type: NodePort
+      exposeStatus: true
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply examples/basic/tidb-cluster.yaml:37]: This is updated by mistake.. I think we don't need to update this file

File: pkg/util/util.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply pkg/util/util.go:244]: Or we may check `eport.NodePort`, if !=0, skip the logic to get port from existedSvc?

[the remark this answers is not in the document]
Comment (lichunzhu) [reply pkg/util/util.go]: nice advice, updated in https://github.com/pingcap/tidb-operator/pull/2941/commits/2a9b38ebb1445589e57618e358e536b1fe471ab0

File: examples/basic/tidb-cluster.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lichunzhu) [reply examples/basic/tidb-cluster.yaml:37]: updated in https://github.com/pingcap/tidb-operator/pull/2941/commits/2a9b38ebb1445589e57618e358e536b1fe471ab0

[the remark this answers is not in the document]
Comment (cofyc) [reply examples/basic/tidb-cluster.yaml:37]: IMO this is not necessary for the basic example

[the remark this answers is not in the document]
Comment (lichunzhu) [reply examples/basic/tidb-cluster.yaml]: deleted in https://github.com/pingcap/tidb-operator/pull/2941/commits/da93fcb4a048ac407cb0a205fc2e4cc6d2337d6d

**Pull request merged**